src/components/Avatar/index.js

```diff
@@ -47,22 +49,40 @@ const Avatar = ({
     return returnInitials.slice(0, 2);
   }, [name]);
 
-  const image = useMemo(() => {
-    if (!imageUrl) return null;
-    // Only load the image if the url response is 200
-    try {
-      const http = new XMLHttpRequest();
-      http.open('HEAD', imageUrl, false);
-      http.send();
-      return http.status === 200 ? imageUrl : null;
-    } catch (e) {
-      return null;
-    }
+  useEffect(() => {
+    if (!imageUrl) return () => {};
+
+    setLoadedImageUrl(null);
+
+    const img = new Image();
```

Comment:
I would have probably put the Image in a `ref` and used `removeEventListener` and a named function for the event handler.

But I can't think of a reason this wouldn't work, unless the anonymous function's closure does something weird with the state setter `setLoadedImageUrl`. I'll approve it, as long as you'll keep an eye out for weird side effects.

Maybe add a unit test for the case where the `imageUrl` changes after mounting the component?

Reply:
Didn't quite get the ref solution but I'm taking care of weird state updates (caused by race conditions) by removing the `onload` and `onerror` callbacks in the cleanup function every time `imageUrl` changes. I can add a new test case, but I'll need to add two new props `onLoad` and `onError`, so it can be easily tested 🙂

Reply:
I'm not sure how to test it, I'll need to add a delay so I can change the `imageUrl` prop before `onload` gets called, any idea on how to test that?

Reply:
I think you're ok as it is. I can see that being a pain to test.
You were at least able to validate in storybook, right?

Reply:
I did validate it in Storybook. Anyway, I figured out a way of testing it, let me know if you like it, otherwise, we revert it.

Reply:
cc: @pixelbandito